test/source-sink-errors/mzcompose.py

```diff
@@ -540,12 +540,25 @@ def assert_recovery(self, c: Composition) -> None:
 ]
 
 
-def workflow_default(c: Composition) -> None:
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     """Test the detection and reporting of source/sink errors by
     introducing a Disruption and then checking the mz_internal.mz_*_statuses tables
     """
 
-    sharded_disruptions = buildkite.shard_list(disruptions, lambda s: s.name)
+    parser.add_argument("disruptions", nargs="*", default=[d.name for d in disruptions])
+
+    args = parser.parse_args()
+
+    to_run = []
+    for name in args.disruptions:
+        for disruption in disruptions:
+            if disruption.name == name:
+                to_run.append(disruption)
+                break
+        else:
+            raise ValueError(f"Unknown disruption {name}")
+
+    sharded_disruptions = buildkite.shard_list(to_run, lambda s: s.name)
```

Comment:
(Nit: I find the naming slightly confusing `to_run` won't be the ones that run but ok.)